android-o/AndroidPlayLocation/LocationUpdatesPendingIntent/LocationUpdatesPendingIntent/MainActivity.cs

```diff
@@ -0,0 +1,304 @@
+			UpdateButtonsState(Utils.GetRequestingLocationUpdates(this));
+			mLocationUpdatesResultView.Text = Utils.GetLocationUpdatesResult(this);
+		}
+
+		protected override void OnStop()
+		{
+			PreferenceManager.GetDefaultSharedPreferences(this).UnregisterOnSharedPreferenceChangeListener(this);
+			base.OnStop();
+		}
+
+		/**
+	     * Sets up the location request. Android has two location request settings:
+	     * {@code ACCESS_COARSE_LOCATION} and {@code ACCESS_FINE_LOCATION}. These settings control
+	     * the accuracy of the current location. This sample uses ACCESS_FINE_LOCATION, as defined in
+	     * the AndroidManifest.xml.
+	     * <p/>
+	     * When the ACCESS_FINE_LOCATION setting is specified, combined with a fast update
+	     * interval (5 seconds), the Fused Location Provider API returns location updates that are
+	     * accurate to within a few feet.
+	     * <p/>
+	     * These settings are appropriate for mapping applications that show real-time location
+	     * updates.
+	     */
+		void CreateLocationRequest()
+		{
+			mLocationRequest = new LocationRequest();
+
+			// Sets the desired interval for active location updates. This interval is
+			// inexact. You may not receive updates at all if no location sources are available, or
+			// you may receive them slower than requested. You may also receive updates faster than
+			// requested if other applications are requesting location at a faster interval.
+			// Note: apps running on "O" devices (regardless of targetSdkVersion) may receive updates
+			// less frequently than this interval when the app is no longer in the foreground.
+			mLocationRequest.SetInterval(UpdateInterval);
+
+			// Sets the fastest rate for active location updates. This interval is exact, and your
+			// application will never receive updates faster than this value.
+			mLocationRequest.SetFastestInterval(FastestUpdateInterval);
+
+			mLocationRequest.SetPriority(LocationRequest.PriorityHighAccuracy);
+
+			// Sets the maximum time when batched location updates are delivered. Updates may be
+			// delivered sooner than this interval.
+			mLocationRequest.SetMaxWaitTime(MaxWaitTime);
+		}
+
+		PendingIntent GetPendingIntent()
+		{
+			// Note: for apps targeting API level 25 ("Nougat") or lower, either
+			// PendingIntent.getService() or PendingIntent.getBroadcast() may be used when requesting
+			// location updates. For apps targeting API level O, only
+			// PendingIntent.getBroadcast() should be used. This is due to the limits placed on services
+			// started in the background in "O".
+
+			// TODO(developer): uncomment to use PendingIntent.GetService().
+			//        var intent = new Intent(this, typeof(LocationUpdatesIntentService));
+			//        intent.SetAction(LocationUpdatesIntentService.ActionProcessUpdate);
+			//        return PendingIntent.GetService(this, 0, intent, PendingIntentFlags.UpdateCurrent);
+
+			var intent = new Intent(this, typeof(LocationUpdatesIntentService));
```

Comment:
I believe this should use `typeof(LocationUpdatesBroadcastReceiver)`, rather than `LocationUpdatesIntentService`

Reply:
Done